src/gt4py/next/otf/binding/nanobind.py

```diff
@@ -89,7 +89,7 @@ class BindingFile(eve.Node):
 
 
 def _type_string(type_: ts.TypeSpec) -> str:
-    if isinstance(type_, ts.TupleType):
+    if isinstance(type_, (ts.TupleType, ts.NamedCollectionType)):
```

Comment:
Actually, I think this kind of instance checks should  use `type_specifications.COLLECTION_TYPE_SPECS` everywhere